qcodes/dataset/__init__.py

```diff
@@ -41,7 +41,7 @@
     datasaver_builder,
     dond_into,
 )
-from .measurements import Measurement
+from .measurements import Measurement, str_or_register_name
```

Comment:
what is the purpose of this function being public? I'd think that it's very specific to the implementation of the Measurement class, and does not need to be used outside at all.